vortex-layout/src/layouts/flat/reader.rs

```diff
@@ -43,11 +48,16 @@ impl FlatReader {
         name: Arc<str>,
         segment_source: Arc<dyn SegmentSource>,
     ) -> Self {
+        let len = layout
+            .row_count
```

Comment:
Change FlatLayout struct to hold row_count as usize, since it's guaranteed to fit into one. Then can remove this stuff.